etcdmgmt/etcd-mgmt.go

```diff
@@ -1,24 +1,38 @@
 package etcdmgmt
 
 import (
+	"bufio"
 	"encoding/json"
 	"io/ioutil"
 	"net/http"
 	"os"
 	"os/exec"
 	"strconv"
+	"strings"
 	"syscall"
 	"time"
 
+	"github.com/gluster/glusterd2/context"
 	"github.com/gluster/glusterd2/utils"
 
 	log "github.com/Sirupsen/logrus"
 )
 
 // ExecName is to indicate the executable name, useful for mocking in tests
-var ExecName = "etcd"
+var (
+	ExecName           = "etcd"
+	listenClientUrls   = "http://" + context.Hostname + ":2379"
```

Comment:
Instead of context.Hostname I suggest we should use utils.GetLocalIP() where the function should be defined as following in utils.go:

+func GetLocalIP() string {
+       addrs, err := net.InterfaceAddrs()
+       if err != nil {
+               log.Error(err)
+               return ""
+       }
+       log.Info(addrs)
+       for _, address := range addrs {
+               // check the address type and if it is not a loopback the display it
+               if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+                       if ipnet.IP.To4() != nil {
+                               return ipnet.IP.String()
+                       }
+               }
+       }
+
+       return ""
+}